### PR TITLE
Enable TreatWarningsAsErrors for MediaBrowser.Model

### PR DESCRIPTION
**Pull request by Bond-009**
#2149

### CODE REVIEW

File: MediaBrowser.Model/Dlna/StreamInfo.cs [hunk 1]
```diff
@@ -152,887 +113,928 @@ public string GetOption(string name)
...
             {
-                throw new ArgumentNullException(nameof(baseUrl));
+                if (MediaSource != null)
+                {
+                    return MediaSource.VideoStream;
+                }
+
+                return null;
             }
+        }
```
Comment (barronpm) [inline MediaBrowser.Model/Dlna/StreamInfo.cs:146]: `public MediaStream TargetVideoStream => MediaSource?.VideoStream;`

File: MediaBrowser.Model/Dlna/StreamInfo.cs [hunk 2]
```diff
@@ -152,887 +113,928 @@ public string GetOption(string name)
...
             {
-                return MediaSource.Path;
+                if (MediaSource != null)
+                {
+                    return MediaSource.GetDefaultAudioStream(AudioStreamIndex);
+                }
+
+                return null;
             }
+        }
```
Comment (barronpm) [inline MediaBrowser.Model/Dlna/StreamInfo.cs:130]: `public MediaStream TargetAudioStream => MediaSource?.GetDefaultAudioStream(AudioStreamIndex);`

**Comment by Bond-009** [comment]
1672 Warnings left after this PR

File: MediaBrowser.Model/LiveTv/LiveTvChannelQuery.cs [hunk 3]
```diff
@@ -48,13 +54,13 @@ public class LiveTvChannelQuery
         public Guid UserId { get; set; }
 
         /// <summary>
-        /// Skips over a given number of items within the results. Use for paging.
+        /// gets or sets the start index. Used for paging.
```
Comment (crobibero) [inline MediaBrowser.Model/LiveTv/LiveTvChannelQuery.cs:57]:
```suggestion
        /// Gets or sets the start index. Used for paging.
```

**Review by crobibero — APPROVED** [review]

**Review by barronpm — APPROVED** [review]

**Pull request merged**